build.mjs

```diff
@@ -0,0 +1,179 @@
+}
+
+/**
+ * Build a component set zip for path.
+ * @param {string} componentSetPath
+ */
+async function buildComponentSet(componentSetPath) {
+    console.log(`Building component set "${componentSetPath}"`);
+    await generateDesignFile(componentSetPath);
+    await compileDesignFile(componentSetPath);
+    await generateVendorScript(componentSetPath);
+    await validateComponentSet(componentSetPath);
+    await buildComponentSetZip(componentSetPath);
+}
+
+/**
+ * Runs component set validation and throws error when something is wrong.
+ * @param {string} componentSetPath
+ */
+export async function validateComponentSet(componentSetPath) {
+    console.log(`Validating component set "${componentSetPath}"`);
+    const valid = await validateFolder(componentSetPath);
+    if (!valid) {
+        throw new Error(`Package failed validation: "${componentSetPath}". See errors above.`);
+    } else {
+        console.log(`Component set "${componentSetPath}" is valid!`);
+    }
+}
+
+/**
+ * Generates design.scss from style files.
+ * @param {string} componentSetPath
+ */
+async function generateDesignFile(componentSetPath) {
+    console.log(`Generating design.scss for component set "${componentSetPath}"`);
+    try {
+        const definition = await getComponentDefinition(componentSetPath);
+        const componentNames = definition.components.map((compDef) => compDef.name);
+
+        let content = `/*
+* This file has been generated while building the components package.
+* PLEASE DO NOT MODIFY THIS FILE BY HAND.
+*/\n${buildScssString('common')}`;
+
+        for (const componentName of componentNames) {
+            content += buildScssString(componentName);
+        }
+
+        await fs.writeFile(path.join(componentSetPath, 'styles/design.scss'), content);
+    } catch (e) {
+        // Don't fail task, but let the components validator generate the actual error,
+        // in case the component definition is malformed.
+        console.error('Failed to generate design.scss file: \n', e);
+        return;
+    }
+}
+/**
+ * @param {string} componentSetPath
+ */
+async function getComponentDefinition(componentSetPath) {
+    return JSON.parse(await fs.readFile(path.join(componentSetPath, 'components-definition.json'), 'utf8'));
+}
+
+/**
+ * @param {string} componentName
+ * @returns {string}
+ */
+function buildScssString(componentName) {
+    return `@import "${componentName}";\n`;
+}
+
+/**
+ * Compiles design.scss.
+ * @param {string} componentSetPath
+ */
+async function compileDesignFile(componentSetPath) {
+    console.log(`Compiling design.scss for component set "${componentSetPath}"`);
+    const result = sass.compile(path.join(componentSetPath, 'styles/design.scss'));
+    await fs.writeFile(path.join(componentSetPath, 'styles/design.css'), result.css);
+}
+
+/**
+ * Generates vendor script by concatenating and uglifying the result.
+ * @param {string} componentSetPath
+ */
+async function generateVendorScript(componentSetPath) {
```

Comment:
Code in this function is specific to the component set and should not be here I guess? Should we allow component set specific build steps?

Reply:
Makes sense, but I wanted to leave it out of the PR. I'm not sure what to do with the scripts yet.

Reply:
We can do it in steps indeed.